packages/data/src/components/use-select/index.js

```diff
@@ -20,28 +20,36 @@ import useAsyncMode from '../async-mode-provider/use-async-mode';
 const noop = () => {};
 const renderQueue = createQueue();
 
-/** @typedef {import('../../types').StoreDescriptor} StoreDescriptor */
+/**
+ * @typedef {import('../../types').StoreDescriptor<C>} StoreDescriptor
+ * @template C
+ */
+/**
+ * @typedef {import('../../types').ReduxStoreConfig<State,Actions,Selectors>} ReduxStoreConfig
+ * @template State,Actions,Selectors
+ */
+/**
+ * @typedef {import('../../types').UseSelectReturn<T>} UseSelectReturn
+ * @template T
+ */
+/** @typedef {import('../../types').MapSelect} MapSelect */
 
 /**
  * Custom react hook for retrieving props from registered selectors.
  *
  * In general, this custom React hook follows the
  * [rules of hooks](https://reactjs.org/docs/hooks-rules.html).
  *
- * @param {Function|StoreDescriptor|string} mapSelect Function called on every state change. The
- *                                                    returned value is exposed to the component
- *                                                    implementing this hook. The function receives
- *                                                    the `registry.select` method on the first
- *                                                    argument and the `registry` on the second
- *                                                    argument.
- *                                                    When a store key is passed, all selectors for
- *                                                    the store will be returned. This is only meant
- *                                                    for usage of these selectors in event
- *                                                    callbacks, not for data needed to create the
- *                                                    element tree.
- * @param {Array}                           deps      If provided, this memoizes the mapSelect so the
- *                                                    same `mapSelect` is invoked on every state
- *                                                    change unless the dependencies change.
+ * @template {MapSelect | StoreDescriptor<any>} T
+ * @param {T} mapSelect Function called on every state change. The returned value is
+ * 											exposed to the component implementing this hook. The function
+ * 											receives the `registry.select` method on the first argument
+ * 											and the `registry` on the second argument.
+ *                      When a store key is passed, all selectors for the store will be
+ *                      returned. This is only meant for usage of these selectors in event
+ *                      callbacks, not for data needed to create the element tree.
+ * @param {*} deps      If provided, this memoizes the mapSelect so the same `mapSelect` is
```

Comment:
What is the intention behind the `*` type? It's not a standard TS type annotation. It maps to `any` or something like that right? Could we use the proper TS type instead, if possible?

Should this actually be like an array of unknown or something like that?

Reply:
Great spot! I just updated that to unknown[]